library/src/com/google/maps/android/kml/KmlParser.java

```diff
@@ -0,0 +1,127 @@
+
+    private final static String STYLE_MAP = "StyleMap";
+
+    private final static String PLACEMARK = "Placemark";
+
+    private final static String GROUND_OVERLAY = "GroundOverlay";
+
+    private final static String CONTAINER_REGEX = "Folder|Document";
+
+    private final KmlStyleParser styleParser;
+
+    private final KmlFeatureParser featureParser;
+
+    private final KmlContainerParser containerParser;
+
+    private final XmlPullParser mParser;
+
+    private final HashMap<KmlPlacemark, Object> mPlacemarks;
+
+    private final ArrayList<KmlContainer> mFolders;
+
+    private final HashMap<String, KmlStyle> mStyles;
+
+    private final HashMap<KmlGroundOverlay, GroundOverlay> mGroundOverlays;
+
+    /**
+     * Creates a new KmlParser object
+     *
+     * @param parser parser containing the KML file to parse
+     */
+    /* package */ KmlParser(XmlPullParser parser) {
+        mParser = parser;
+        mPlacemarks = new HashMap<KmlPlacemark, Object>();
+        mFolders = new ArrayList<KmlContainer>();
+        mStyles = new HashMap<String, KmlStyle>();
+        styleParser = new KmlStyleParser(mParser);
+        featureParser = new KmlFeatureParser(mParser);
+        containerParser = new KmlContainerParser(mParser);
+        mGroundOverlays = new HashMap<KmlGroundOverlay, GroundOverlay>();
+    }
+
+    /**
+     * Parses the KML file and stores the created KmlStyle and KmlPlacemark
+     */
+    /* package */ void parseKml() throws XmlPullParserException, IOException {
+        int eventType = mParser.getEventType();
+        while (eventType != XmlPullParser.END_DOCUMENT) {
+            if (eventType == XmlPullParser.START_TAG) {
+                if (mParser.getName().matches(CONTAINER_REGEX)) {
+                    containerParser.createContainer();
+                    mFolders.add(containerParser.getContainer());
+                }
+                if (mParser.getName().equals(STYLE)) {
+                    styleParser.createStyle();
+                    mStyles.put(styleParser.getStyle().getStyleId(), styleParser.getStyle());
+                }
+                if (mParser.getName().equals(STYLE_MAP)) {
+                    styleParser.createStyleMap();
+                }
+                if (mParser.getName().equals(PLACEMARK)) {
+                    featureParser.createPlacemark();
+                    mPlacemarks.put(featureParser.getPlacemark(), null);
+                }
+                if (mParser.getName().equals(GROUND_OVERLAY)) {
+                    featureParser.createGroundOverlay();
+                    mGroundOverlays.put(featureParser.getGroundOverlay(), null);
+                }
+            }
+            eventType = mParser.next();
+        }
+    }
+
+    /**
+     * @return List of styles created by the parser
+     */
+    /* package */ HashMap<String, KmlStyle> getStyles() {
+        //TODO: Need to put an empty new style, can probably be put somewhere better
```

Comment:
So I see it's because it's a default style?  Might be worth having a separate field for default style?